gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -269,11 +277,31 @@ public GoogleCloudStorageImpl(GoogleCloudStorageOptions options, Storage gcs) {
     if (gcs.getRequestFactory() != null) {
       this.httpRequestInitializer = gcs.getRequestFactory().getInitializer();
     }
+
+    initAuthorizationProvider(storageOptions.getAuthorizationProvider(),
+            storageOptions.getAuthorizationProviderInitArgument());
   }
 
   @VisibleForTesting
   protected GoogleCloudStorageImpl() {
     this.storageOptions = GoogleCloudStorageOptions.builder().setAppName("test-app").build();
+    initAuthorizationProvider("", "");
+  }
+
+  private void initAuthorizationProvider(String className, String initArg) {
+    if (! "".equals(storageOptions.getAuthorizationProvider())) {
+      try {
```

Comment:
Move the reflection logic into `GoogleHadoopFileSystemConfiguration.getGcsOptionsBuilder` method, and replace `init` with `setConf` from the `Configurable` interface.